src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,14 +35,18 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+        public OrderResponseDto completeOrder(Authentication authentication) {
+        User user = userService.findByEmail(authentication.getName()).orElseThrow(() ->
+                    new RuntimeException("Can't find user by email " + authentication.getName()));
```

Comment:
It's better to throw a custom exception like `EntityNotFoundException` or `DataProcessingException`.